python/setup.py

```diff
@@ -266,7 +266,14 @@ def get_packages(self):
             "fastapi",
             "watchfiles",
         ],
-        "tune": ["pandas", "tensorboardX>=1.9", "requests", pyarrow_dep, "fsspec"],
+        "tune": [
+            "pandas",
+            "tensorboardX>=1.9",
+            "requests",
+            pyarrow_dep,
+            "fsspec",
+            "pydantic!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.*,!=2.4.*,<3",
```

Comment:
I don't think we need `pydantic` as a dependency of `train`, just `default`.

Reply:
I am planning to put the pydantic class under `ray/train/_internal`. If we don't have it for `train`, are we still be able to build `ray[train]`?

Reply:
The code should just be guarded by whether or not `pydantic` is installed. Similar to `torch`. 😄